### ISSUE TITLE
[Component] [date-picker] date-picker input-box display value with localization format

### ISSUE THREAD

**Issue by jongmin4943** [element-plus/element-plus#9158]
Bug Type: **`Component`**

## Environment

- Vue Version: `3.2.31`
- Element Plus Version: `2.2.12`
- Browser / OS: `Chrome / Window`
- Build Tool: `Vite`

### Related Component

- `el-date-picker`

### Reproduction Link

[Element Plus Playground](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuaW1wb3J0IHsgdmVyc2lvbiBhcyBFcFZlcnNpb24gfSBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBFbGVtZW50UGx1cyB9IGZyb20gJ0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlJ1xuaW1wb3J0IHsgRWxDb25maWdQcm92aWRlciB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB6aENuIGZyb20gJ2VsZW1lbnQtcGx1cy9lcy9sb2NhbGUvbGFuZy96aC1jbidcblxuY29uc3QgdmFsdWUxID0gcmVmKCcnKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPGVsLWNvbmZpZy1wcm92aWRlciA6bG9jYWxlPVwiemhDblwiPlxuICAgIDxlbC1kYXRlLXBpY2tlclxuICAgICAgICB2LW1vZGVsPVwidmFsdWUxXCJcbiAgICAgICAgdHlwZT1cImRhdGVcIlxuICAgICAgICBwbGFjZWhvbGRlcj1cIlBpY2sgYSBEYXRlXCJcbiAgICAgICAgZm9ybWF0PVwiWVlZWS9NTS9ERCBkZGRcIlxuICAgICAgLz5cbiAgPC9lbC1jb25maWctcHJvdmlkZXI+XG5cdFx0XHRcbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnRfbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7fVxufSIsIl9vIjp7fX0=)

### Steps to reproduce

```vue
<script setup lang="ts">
import { ref } from 'vue'
import { ElementPlus } from '@element-plus/icons-vue'
import { ElConfigProvider } from 'element-plus'
import ko from 'element-plus/es/locale/lang/ko'

const value1 = ref('')
</script>

<template>
  <el-config-provider :locale="ko">
    <el-date-picker
        v-model="value1"
        type="date"
        placeholder="Pick a Date"
        format="YYYY/MM/DD ddd"
      />
  </el-config-provider>	
</template>
```

### What is Expected?

date-picker input-box shows YYYY/MM/DD ddd with Korean

### What is actually happening?

date-picker input-box shows YYYY/MM/DD ddd with English

### Additional comments

![22](https://user-images.githubusercontent.com/77711322/182306498-c734affc-606b-4bfa-848d-5e17a47064a6.jpeg)

**Comment by ryuhangyeong** [comment]
## the most fragmentary method

As a result of testing in Vue-cli, I could do what you wanted by installing dayjs and importing locale.

![image](https://user-images.githubusercontent.com/27342882/182369783-79b162aa-f606-4a5d-808a-a87435b479f2.png)

```js
import { createApp } from 'vue'
import ElementPlus from 'element-plus'
import 'element-plus/dist/index.css'
import App from './App.vue'

import 'dayjs/locale/ko' // here

const app = createApp(App)

app.use(ElementPlus)
app.mount('#app')
```

**But this is not a fundamental solution.**

When I looked at the code base of element-plus, I checked that it will run normally if it is done as below.

However, it is very inefficient to bring locale like this, so I think we need to find another way.

```js
// element-plus/packages/components/time-picker/index.ts

import TimePicker from './src/time-picker'
import CommonPicker from './src/common/picker.vue'
import TimePickPanel from './src/time-picker-com/panel-time-pick.vue'

import type { App } from 'vue'
import type { SFCWithInstall } from '@element-plus/utils'

import 'dayjs/locale/zh-cn' // here
import 'dayjs/locale/ko' // here
import 'dayjs/locale/en' // here

export * from './src/utils'
export * from './src/constants'
export * from './src/common/props'

const _TimePicker = TimePicker as SFCWithInstall<typeof TimePicker>

_TimePicker.install = (app: App) => {
  app.component(_TimePicker.name, _TimePicker)
}

export { CommonPicker, TimePickPanel }
export default _TimePicker
export const ElTimePicker = _TimePicker
```

![image](https://user-images.githubusercontent.com/27342882/182370579-611a64dc-4c89-4e2d-af04-618a4745d5b5.png)

I'll try to find a better solution. For now, you can solve the problem as a temporary measure!

**Comment by ryuhangyeong** [comment]
## temporary measure

It's not a fundamental solution, but as a workaround, I was able to do what you wanted using the method below.

![image](https://user-images.githubusercontent.com/27342882/182376219-55c75b8f-e5be-4df5-89b2-a6f337b65879.png)

```js
import { createApp } from 'vue'
import ElementPlus from 'element-plus'
import 'element-plus/dist/index.css'
import App from './App.vue'

import 'dayjs/locale/ko' // here

const app = createApp(App)

app.use(ElementPlus)
app.mount('#app')
```

- vue cli
- vue3, vite

**Comment by jongmin4943** [comment]
That problem appeared after I updated latest version of Element-plus. (Before it was 2.0.2 and no problem.) My project already had dayjs with ko locale setup but it wasn't working after update. I made new project and set up similar environments and tried your solution and it's working. I don't know what the problem is. I'm gonna just use with English for now and will wait for fixing bug!

**Comment by ryuhangyeong** [comment]
@jongmin4943 Let's look at the issues by going through the history of our previous work.

**Comment by jongmin4943** [comment]
I figured it out. It was dayjs version issue. Original project had dayjs '1.10.7' and element-plus '2.0.2'.
I only upgrade element-plus to latest and the problem came out. I was trying many things to fix it and finally it got solved by changing dayjs version from '1.10.7' to latest. Thanks!
